backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendFactory.java

```diff
@@ -87,15 +89,51 @@ public BackendImplementor create(EventContext eventContext, BackendBuildContext
 		try {
 			threads = new BackendThreads( eventContext.render() );
 
+			// First, let's see if the factory was configured explicitly:
 			Optional<BeanHolder<? extends ElasticsearchClientFactory>> customClientFactoryHolderOptional =
 					CLIENT_FACTORY.getAndMap( propertySource, beanResolver::resolve );
 			if ( customClientFactoryHolderOptional.isPresent() ) {
 				clientFactoryHolder = customClientFactoryHolderOptional.get();
-				ConfigurationLog.INSTANCE.backendClientFactory( clientFactoryHolder, eventContext.render() );
 			}
 			else {
-				clientFactoryHolder = BeanHolder.of( new ElasticsearchClientFactoryImpl() );
+				// otherwise let's find all client factories and pick
+				List<BeanReference<ElasticsearchClientFactory>> clientFactoryReferences =
+						beanResolver.allConfiguredForRole( ElasticsearchClientFactory.class );
+				if ( clientFactoryReferences.isEmpty() ) {
+					throw ConfigurationLog.INSTANCE.backendClientFactoryNotConfigured( eventContext.render() );
+				}
+				// if there's just one -- use it:
+				else if ( clientFactoryReferences.size() == 1 ) {
+					clientFactoryHolder = clientFactoryReferences.get( 0 ).resolve( beanResolver );
+				}
+				// if there are 2 of them, maybe one is the "default" one, if so -- use the other one
+				else if ( clientFactoryReferences.size() == 2 ) {
+					try {
+						var defaultFactoryReference = beanResolver.namedConfiguredForRole( ElasticsearchClientFactory.class )
+								.get( ElasticsearchClientFactory.DEFAULT_BEAN_NAME );
+
+						var first = clientFactoryReferences.get( 0 );
+						var second = clientFactoryReferences.get( 1 );
+						if ( first == defaultFactoryReference ) {
+							clientFactoryHolder = second.resolve( beanResolver );
+						}
+						else if ( second == defaultFactoryReference ) {
+							clientFactoryHolder = first.resolve( beanResolver );
+						}
+					}
+					catch (SearchException e) {
+						// ignore and fail later;
+					}
```

Comment:
Why not just propagate the exception?

This kind of swallowing of exceptions is pretty hard to debug...